### PR TITLE
🌱 Use patcher in operator to replace previous update

### PR DESCRIPTION
**Pull request by qiujian16**
## Related issue(s)

Fixes #

### CODE REVIEW

File: pkg/operator/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go [hunk 1]
```diff
@@ -272,45 +274,30 @@ func (n *klusterletController) sync(ctx context.Context, controllerContext facto
...
 	if len(errs) == 0 {
-		appliedCondition = &metav1.Condition{
+		meta.SetStatusCondition(&klusterlet.Status.Conditions, metav1.Condition{
 			Type: klusterletApplied, Status: metav1.ConditionTrue, Reason: "KlusterletApplied",
-			Message: "Klusterlet Component Applied"}
+			Message: "Klusterlet Component Applied"})
 	} else {
-		if appliedCondition == nil {
-			appliedCondition = &metav1.Condition{
+		if meta.FindStatusCondition(klusterlet.Status.Conditions, klusterletApplied) == nil {
```
Comment (zhujian7) [inline pkg/operator/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go:284]: it seems this check is not necessary `if meta.FindStatusCondition(klusterlet.Status.Conditions, klusterletApplied) == nil {`?
Think about if a klusterlet upgraded to a new version and applied failed, the old version's klusterletApplied exists, the new version applied failed error will not be recorded.

File: pkg/operator/operators/klusterlet/controllers/klusterletcontroller/klusterlet_managed_reconcile.go [hunk 2]
```diff
@@ -189,12 +195,7 @@ func (r *managedReconcile) cleanUpAppliedManifestWorks(ctx context.Context, klus
 		}
 
 		// remove finalizer if exists
-		if mutated := removeFinalizer(&appliedManifestWorks.Items[index], appliedManifestWorkFinalizer); !mutated {
-			continue
-		}
-
-		_, err := r.managedClusterClients.appliedManifestWorkClient.Update(ctx, &appliedManifestWorks.Items[index], metav1.UpdateOptions{})
-		if err != nil && !errors.IsNotFound(err) {
+		if err := patcher.RemoveFinalizer(ctx, &appliedManifestWorks.Items[index], appliedManifestWorkFinalizer); err != nil && !errors.IsNotFound(err) {
```
Comment (zhujian7) [inline pkg/operator/operators/klusterlet/controllers/klusterletcontroller/klusterlet_managed_reconcile.go:198]: when will we get a "IsNotFound" error?

Comment (qiujian16) [reply pkg/operator/operators/klusterlet/controllers/klusterletcontroller/klusterlet_managed_reconcile.go:198]: when the resource has been deleted.

Comment (zhujian7) [reply pkg/operator/operators/klusterlet/controllers/klusterletcontroller/klusterlet_managed_reconcile.go:198]: do we need to check this error in the common patcher package?

Comment (qiujian16) [reply pkg/operator/operators/klusterlet/controllers/klusterletcontroller/klusterlet_managed_reconcile.go:198]: yeah good point

**Comment by zhujian7** [comment]
LGTM
/assign @zhiweiyin318

**Pull request merged**